File: lib/constants/app_constants.ts
// Color Palette - OKLCH Color Space
export const colors = {
  // Light Mode
  light: {
    background: 'oklch(0.98 0.002 240)',
    foreground: 'oklch(0.15 0.01 240)',
    card: 'oklch(0.97 0.002 240)',
    cardForeground: 'oklch(0.15 0.01 240)',
    primary: 'oklch(0.35 0.01 240)',
    primaryForeground: 'oklch(0.97 0.002 240)',
    secondary: 'oklch(0.65 0.01 240)',
    secondaryForeground: 'oklch(0.15 0.01 240)',
    muted: 'oklch(0.88 0.005 240)',
    mutedForeground: 'oklch(0.45 0.01 240)',
    accent: 'oklch(0.35 0.01 240)',
    accentForeground: 'oklch(0.97 0.002 240)',
    destructive: 'oklch(0.55 0.2 25)',
    destructiveForeground: 'oklch(0.97 0.002 240)',
    border: 'oklch(0.92 0.005 240)',
    input: 'oklch(0.92 0.005 240)',
    ring: 'oklch(0.45 0.01 240)',
    likeRed: '#ef4444', // red-500
  },

  // Dark Mode
  dark: {
    background: 'oklch(0.145 0 0)',
    foreground: 'oklch(0.985 0 0)',
    card: 'oklch(0.145 0 0)',
    cardForeground: 'oklch(0.985 0 0)',
    primary: 'oklch(0.985 0 0)',
    primaryForeground: 'oklch(0.205 0 0)',
    secondary: 'oklch(0.269 0 0)',
    secondaryForeground: 'oklch(0.985 0 0)',
    muted: 'oklch(0.269 0 0)',
    mutedForeground: 'oklch(0.708 0 0)',
    accent: 'oklch(0.269 0 0)',
    accentForeground: 'oklch(0.985 0 0)',
    destructive: 'oklch(0.396 0.141 25.723)',
    destructiveForeground: 'oklch(0.637 0.237 25.331)',
    border: 'oklch(0.269 0 0)',
    input: 'oklch(0.269 0 0)',
    ring: 'oklch(0.439 0 0)',
    likeRed: '#ef4444',
  },
};

// Typography
export const typography = {
  fontFamily: {
    sans: "'Geist', 'Geist Fallback'",
    mono: "'Geist Mono', 'Geist Mono Fallback'",
  },
  fontSize: {
    xs: '12px', // 0.75rem
    sm: '14px', // 0.875rem
    base: '16px', // 1rem
    lg: '18px', // 1.125rem
    xl: '20px', // 1.25rem
    '2xl': '24px', // 1.5rem
    '3xl': '30px', // 1.875rem
    '4xl': '36px', // 2.25rem
    '5xl': '48px', // 3rem
    '6xl': '60px', // 3.75rem
  },
  fontWeight: {
    regular: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },
  lineHeight: {
    tight: 1.2,
    normal: 1.4,
    relaxed: 1.6,
  },
};

// Spacing (in pixels)
export const spacing = {
  0: '0px',
  1: '4px',
  2: '8px',
  4: '16px',
  6: '24px',
  8: '32px',
  12: '48px',
  16: '64px',
};

// Border Radius
export const borderRadius = {
  xs: '2px',
  sm: '4px',
  md: '6px',
  lg: '8px',
  xl: '12px',
  full: '9999px',
};

// Responsive Breakpoints (pixels)
export const breakpoints = {
  mobile: 320,
  sm: 640,
  md: 768,
  lg: 1024,
  xl: 1280,
};

// Animation/Transition Durations (milliseconds)
export const animations = {
  fast: 150,
  normal: 300,
  slow: 500,
};

// Z-Index Scale
export const zIndex = {
  hide: -1,
  auto: 0,
  base: 0,
  navbar: 10,
  dropdown: 50,
  modal: 100,
  topmost: 1000,
};

// UI Constants
export const ui = {
  navbarHeight: 64,
  modalMaxWidth: 672,
  cardMaxWidth: 476,
  streakIncrement: 1,
  defaultStreak: 7,
  autoScrollDebounce: 500,
  autoScrollVelocityThreshold: 8,
};

// API/Share Configuration
export const config = {
  shareAbortErrorName: 'AbortError',
  defaultTheme: 'light',
  enableSystemThemeDetection: true,
};

File: flutter_app/lib/core/constants/app_constants.dart
import 'package:flutter/material.dart';

/// Color Palette - Matching the OKLCH color space from web
class AppColors {
  AppColors._();

  // Light Mode Colors (converted from OKLCH to Hex)
  static const Color lightBackground = Color(0xFFFAFAFA);
  static const Color lightForeground = Color(0xFF1A1A2E);
  static const Color lightCard = Color(0xFFF5F5F5);
  static const Color lightCardForeground = Color(0xFF1A1A2E);
  static const Color lightPrimary = Color(0xFF2D2D3A);
  static const Color lightPrimaryForeground = Color(0xFFF5F5F5);
  static const Color lightSecondary = Color(0xFF6B6B7B);
  static const Color lightSecondaryForeground = Color(0xFF1A1A2E);
  static const Color lightMuted = Color(0xFFDDDDE0);
  static const Color lightMutedForeground = Color(0xFF5A5A6E);
  static const Color lightAccent = Color(0xFF2D2D3A);
  static const Color lightAccentForeground = Color(0xFFF5F5F5);
  static const Color lightDestructive = Color(0xFFDC4545);
  static const Color lightDestructiveForeground = Color(0xFFF5F5F5);
  static const Color lightBorder = Color(0xFFE5E5E8);
  static const Color lightInput = Color(0xFFE5E5E8);
  static const Color lightRing = Color(0xFF5A5A6E);

  // Dark Mode Colors
  static const Color darkBackground = Color(0xFF0A0A0A);
  static const Color darkForeground = Color(0xFFFAFAFA);
  static const Color darkCard = Color(0xFF0A0A0A);
  static const Color darkCardForeground = Color(0xFFFAFAFA);
  static const Color darkPrimary = Color(0xFFFAFAFA);
  static const Color darkPrimaryForeground = Color(0xFF1A1A1A);
  static const Color darkSecondary = Color(0xFF2A2A2A);
  static const Color darkSecondaryForeground = Color(0xFFFAFAFA);
  static const Color darkMuted = Color(0xFF2A2A2A);
  static const Color darkMutedForeground = Color(0xFFA0A0A0);
  static const Color darkAccent = Color(0xFF2A2A2A);
  static const Color darkAccentForeground = Color(0xFFFAFAFA);
  static const Color darkDestructive = Color(0xFF7F1D1D);
  static const Color darkDestructiveForeground = Color(0xFFDC4545);
  static const Color darkBorder = Color(0xFF2A2A2A);
  static const Color darkInput = Color(0xFF2A2A2A);
  static const Color darkRing = Color(0xFF5A5A5A);

  // Special Colors
  static const Color likeRed = Color(0xFFEF4444);
  static const Color streakOrange = Color(0xFFF97316);
  static const Color successGreen = Color(0xFF22C55E);
  static const Color warningYellow = Color(0xFFEAB308);
  static const Color infoBlue = Color(0xFF3B82F6);
}

/// Typography constants
class AppTypography {
  AppTypography._();

  static const String fontFamily = 'Geist';
  static const String fontFamilyMono = 'GeistMono';

  // Font Sizes (in logical pixels)
  static const double fontSizeXs = 12.0;
  static const double fontSizeSm = 14.0;
  static const double fontSizeBase = 16.0;
  static const double fontSizeLg = 18.0;
  static const double fontSizeXl = 20.0;
  static const double fontSize2xl = 24.0;
  static const double fontSize3xl = 30.0;
  static const double fontSize4xl = 36.0;
  static const double fontSize5xl = 48.0;
  static const double fontSize6xl = 60.0;

  // Font Weights
  static const FontWeight fontWeightRegular = FontWeight.w400;
  static const FontWeight fontWeightMedium = FontWeight.w500;
  static const FontWeight fontWeightSemibold = FontWeight.w600;
  static const FontWeight fontWeightBold = FontWeight.w700;

  // Line Heights
  static const double lineHeightTight = 1.2;
  static const double lineHeightNormal = 1.4;
  static const double lineHeightRelaxed = 1.6;
}

/// Spacing constants (in logical pixels)
class AppSpacing {
  AppSpacing._();

  static const double sp0 = 0.0;
  static const double sp1 = 4.0;
  static const double sp2 = 8.0;
  static const double sp3 = 12.0;
  static const double sp4 = 16.0;
  static const double sp5 = 20.0;
  static const double sp6 = 24.0;
  static const double sp8 = 32.0;
  static const double sp10 = 40.0;
  static const double sp12 = 48.0;
  static const double sp16 = 64.0;
  static const double sp20 = 80.0;
  static const double sp24 = 96.0;
}

/// Border Radius constants
class AppRadius {
  AppRadius._();

  static const double xs = 2.0;
  static const double sm = 4.0;
  static const double md = 6.0;
  static const double lg = 8.0;
  static const double xl = 12.0;
  static const double xxl = 16.0;
  static const double full = 9999.0;
}

/// Responsive Breakpoints
class AppBreakpoints {
  AppBreakpoints._();

  static const double mobile = 320.0;
  static const double sm = 640.0;
  static const double md = 768.0;
  static const double lg = 1024.0;
  static const double xl = 1280.0;
}

/// Animation/Transition Durations
class AppAnimations {
  AppAnimations._();

  static const Duration fast = Duration(milliseconds: 150);
  static const Duration normal = Duration(milliseconds: 300);
  static const Duration slow = Duration(milliseconds: 500);
}

/// UI Constants
class AppUI {
  AppUI._();

  static const double navbarHeight = 64.0;
  static const double bottomNavHeight = 64.0;
  static const double modalMaxWidth = 672.0;
  static const double cardMaxWidth = 476.0;
  static const int defaultStreak = 7;
  static const int autoScrollDebounce = 500;
  static const double autoScrollVelocityThreshold = 8.0;
}

/// Z-Index Scale
class AppZIndex {
  AppZIndex._();

  static const int hide = -1;
  static const int auto = 0;
  static const int base = 0;
  static const int navbar = 10;
  static const int dropdown = 50;
  static const int modal = 100;
  static const int topmost = 1000;
}
